src/app/search/page.tsx

```diff
@@ -13,118 +15,124 @@ import {
+        endDate: convertTimestampToDate(content.endDate),
+        organizationName: content.organizationName,
+        categories: content.categories,
+      };
+    }) || [];
+
+  // 데이터 유효성 검사
+  const isValidData = !(isLoading || searchProjects.length === 0 || isError);
 
-  const totalItems = 30; // 총 프로젝트가 30개라고 가정
-  const itemsPerPage = 8; // 한 페이지에 8개씩 보여줌
-  const totalPages = Math.ceil(totalItems / itemsPerPage); // 4
+  // 페이징 관련
+  const totalItems = data?.data.totalCount || DEFAULT_ITEMS_TOTAL_COUNT;
+  const totalPages = Math.ceil(totalItems / ITEMS_PER_PAGE);
 
   const handleClickPaginationItem = (page: number) => {
     setCurrentPage(page);
   };
-  const testData = [
-    {
-      projectId: 1,
-      projectname: 'project.projectName1',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 2,
-      projectname: 'project.proj4ectName2',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 3,
-      projectname: 'project.project1Name',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 5,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 6,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 7,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-    {
-      projectId: 8,
-      projectname: 'project.projectName3',
-      startDate: convertStringToDate('2024-02-02'),
-      endDate: convertStringToDate('2024-05-02'),
-      organizationName: 'project.organizationName',
-      categories: ['기타', '금융', '생산성'],
-      evaluatedMembersCount: 0,
-    },
-  ];
+
+  // "이전" 버튼
+  const handlePrevious = () => {
+    if (currentPage > 1) {
```

Comment:
이렇게 하면 현재 페이지가 1인 경우에도 페이지를 0으로 설정되는거 아닌가요...?
아래처럼 현재 페이지가 0보다 큰 경우에만 페이지를 감소시키고, 현재 페이지가 0인 경우 아무 작업도 하지 않는게 명확할 것 같습니다
```suggestion
    if (currentPage > 0) {
  setCurrentPage((prevPage) => prevPage - 1);
}
```

Reply:
저희 시작 페이지가 0이라서 현재 페이지가 1이라면 0으로 설정해줘야 합니다
0이면 아무것도 안해도 되니, else 문 삭제하겠습니다

Reply:
✅ 0e5b96ceda8262219f047f3878ce120d22648439

수정하다 보니, 새로운 검색 조건이 들어오면 페이징도 초기화 시켜줘야해서 useEffect 구문 추가했습니다!